python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,417 @@
+import copy
+
+from astropy.table import Table
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+from . import CalibCombineTask, CpEfdClient
+from lsst.daf.base import PropertyList, DateTime
+
+__all__ = [
+    "CpFilterScanTask", "CpFilterScanTaskConfig",
+    "CpMonochromatorScanTask", "CpMonochromatorScanConfig"
+]
+
+
+class CpFilterScanConnections(pipeBase.PipelineTaskConnections,
+                              dimensions=("instrument", "detector")):
+    inputExpHandles = cT.Input(
+        name="postISRCCD",
+        doc="Input exposures to measure statistics from.",
+        storageClass="Exposure",
+        dimensions=("instrument", "physical_filter", "exposure", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    inputElectrometerHandles = cT.Input(
+        name="electrometer",
+        doc="Input electrometer data.",
+        storageClass="IsrCalib",
+        dimensions=("instrument", "exposure"),
+        multiple=True,
+        deferLoad=True,
+    )
+    outputData = cT.Output(
+        name="cpFilterScan",
+        doc="Output table to write.",
+        storageClass="ArrowAstropy",
+        dimensions=("instrument", "detector"),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if config and not config.useElectrometer:
+            self.inputs.discard("inputElectrometerHandles")
+
+
+class CpFilterScanTaskConfig(pipeBase.PipelineTaskConfig,
+                             pipelineConnections=CpFilterScanConnections):
+    maskNameList = pexConfig.ListField(
+        dtype=str,
+        doc="Mask list to exclude from statistics calculations.",
+        default=["DETECTED", "BAD", "CR", "NO_DATA", "INTRP"]
+    )
+    useElectrometer = pexConfig.Field(
+        dtype=bool,
+        doc="Use electrometer data?",
+        default=False,
+    )
+    referenceFilter = pexConfig.Field(
+        dtype=str,
+        doc="Filter to use as baseline reference.",
+        default="empty",
+    )
+    combine = pexConfig.ConfigurableField(
+        target=CalibCombineTask,
+        doc="Combine task to use for header merging.",
+    )
+    efdClientInstance = pexConfig.Field(
+        dtype=str,
+        doc="EFD instance to use for monochromator results.",
+        default="usdf_efd",
+    )
+
+
+class CpFilterScanTask(pipeBase.PipelineTask):
+    """Create filter scan from appropriate data.
+
+    This task constructs a filter scan from pairs of flat exposures,
+    one with a filter in the beam and one without a filter.  A
+    monochromator is also included in the beam for both exposures, so
+    that the beam can be limited to a narrow band of wavelengths small
+    than the full bandpass of the filter.
+
+    From these pairs of exposures, we can determine the filter
+    throughput by calculating the flux per second with the filter:
+        F_filter(\lambda0) = median(f_amplifiers) / t_exposure
```

Comment:
These are errors that need to be fixed with either an `r` in front of the triple backtick, or perhaps double backslash?